src/main/java/gov/cabinetoffice/gap/adminbackend/services/UserService.java

```diff
@@ -0,0 +1,30 @@
+package gov.cabinetoffice.gap.adminbackend.services;
+
+import gov.cabinetoffice.gap.adminbackend.repositories.GapUserRepository;
+import gov.cabinetoffice.gap.adminbackend.repositories.GrantApplicantRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final GapUserRepository gapUserRepository;
+
+    private final GrantApplicantRepository grantApplicantRepository;
+
+    public void migrateUser(final String oneLoginSub, final UUID colaSub) {
```

Comment:
Should this method be transactional as we are persisting data across multiple repositories. What are your thoughts?

Reply:
Yeah good shout. If there are errors then reverting changes makes sense, and users would be told to email support at which point we could debug the issue from the original state rather than whatever mess it ended up in